astra-sim/system/collective/DoubleBinaryTreeAllReduce.cc

```diff
@@ -48,7 +48,7 @@ void DoubleBinaryTreeAllReduce::run(EventType event, CallData* data) {
     sim_request snd_req;
     snd_req.srcRank = stream->owner->id;
     snd_req.dstRank = parent;
-    snd_req.tag = stream->stream_id;
+    snd_req.tag = stream->stream_id % this->TAG_OFFSET + this->TAG_OFFSET;
```

Comment:
Think of a way to refactor this so that we won't have to do this operation every time 
(possibility: do this op in front_end_sim_send + add a boolean/enum variable to track whether frontend... is being called by Chakra / Collective / R-Vous